### DISCUSSION TITLE
classes/class_graphedit

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#253] [User-contributed notes]
# classes/class_graphedit

Experimental: This class may be changed or removed in future versions. Inherits: Control< CanvasItem< Node< Object An editor for graph-like structures, using GraphNode s. Description: GraphEdit pro...

https://docs.godotengine.org/en/stable/classes/class_graphedit.html

**Comment by fkeyzuwu** [comment]
To find the actual position of your mouse in the graph edit, considering zoom and scroll_offset, use this:
`var real_position: Vector2 = (mouse_position + scroll_offset) / zoom`

Mouse position can be obtained either from `get_global_mouse_position()` or from `connection_x_empty` signals which provide a `release_position` parameter (they are the same).

If you want to set a `GraphNode`'s position to this, use its `position_offset` property.

**Reply by EdwardChanCH** [reply]
I wish this is part of the official offline doc, so that people can find this formula more easily in the Godot script editor.

**Comment by Lithimlin** [comment]
It is currently not possible to use the `StringName` parameters from signals or connection lists to get the nodes directly via `get_node()`. Instead, the `StringName` needs to first be cast to a `NodePath`. See [this issue](https://github.com/godotengine/godot/issues/71319).

**Comment by admin** [comment]
*Originally posted by [**Gromnax**](https://github.com/Gromnax) on 2026-05-27 10:43:34 UTC*

> For `get_connection_list_from_node(node: StringName)`, it should be noted that autoconnections (a node connected to itself) are reported twice.
> A workaround would be something like this:
>
> ```gdscript
> func get_output_connections_from_node(node:Node, port: int) -> Array:
> 	var retrieved_connections = get_connection_list_from_node(node.name)
> 	var result = []
> 	for connection in retrieved_connections:
> 		var dict = {}
> 		var unique : bool = true
> 		if connection["from_node"] == node.name and connection["from_port"] == port:
> 			dict["node"] = connection["to_node"]
> 			dict["port"] = connection["to_port"]
> 			for other_dic: Dictionary in result:
> 				if other_dic.recursive_equal(dict,2):
> 					unique = false
> 			if unique:
> 				result.push_back(dict)
> 	return result
> ```

**Comment by iAnanich** [comment]
Please, explicitly mention that left ports are only for input, and right ports are only for output connections. Thus `connect_node`'s `from` is asking for index of the slot with right port enabled, and `to` asks for index of the slot with left port enabled.
